locator/token_metadata.cc

```diff
@@ -816,16 +815,14 @@ void token_metadata_impl::calculate_pending_ranges_for_replacing(
     if (_replacing_endpoints.empty()) {
         return;
     }
-    auto address_ranges = strategy.get_address_ranges(unpimplified_this).get0();
     for (const auto& node : _replacing_endpoints) {
         auto existing_node = node.first;
         auto replacing_node = node.second;
+        auto address_ranges = strategy.get_address_ranges(unpimplified_this, existing_node).get0();
         for (const auto& x : address_ranges) {
             seastar::thread::maybe_yield();
-            if (x.first == existing_node) {
```

Comment:
This indicates we should change the signature of get_address_ranges() to return a vector, rather than a multimap with one entry.

Reply:
Or rather, a multimap with identical keys.

Reply:
Yes, and it happens in the second patch (when the function is removed altogether in favor of `get_ranges` which returns a vector.